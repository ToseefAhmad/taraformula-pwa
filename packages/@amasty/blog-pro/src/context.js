import React, { createContext, useContext } from 'react';
import { fullPageLoadingIndicator } from '@magento/venia-ui/lib/components/LoadingIndicator';
import { useBlog } from './talons/useBlog';
import GET_BLOG_INFO_QUERY from './queries/getSettings.graphql';

const AmBlogProContext = createContext();
const { Provider } = AmBlogProContext;

const AmBlogProProvider = props => {
  const { children } = props;

  const talonProps = useBlog({
    query: GET_BLOG_INFO_QUERY
  });

  const { error, loading } = talonProps;

  if (error) {
    if (process.env.NODE_ENV !== 'production') {
      console.error(error);
    }

    console.log('Check if Amasty modules has been installed!');
  }

  if (loading) {
    return fullPageLoadingIndicator;
  }

  const contextValue = {
    ...talonProps
  };

  return <Provider value={contextValue}>{children}</Provider>;
};

export default AmBlogProProvider;

export const useAmBlogProContext = () => useContext(AmBlogProContext);
