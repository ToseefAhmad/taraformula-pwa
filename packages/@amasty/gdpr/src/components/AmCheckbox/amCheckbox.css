.checkbox {
  margin-bottom: 0.625rem;
  margin-left: -0.375rem;
}

.checkout {
  margin: 0.625rem 0 0;
}
