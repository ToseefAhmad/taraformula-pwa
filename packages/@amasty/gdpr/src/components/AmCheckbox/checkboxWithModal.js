import React, { Fragment, Suspense } from 'react';
import { useCheckboxWithModal } from '../../talons/useCheckboxWithModal';
import Checkbox from '@magento/venia-ui/lib/components/Checkbox/checkbox';
import PrivacyModal from '../PrivacyModal/privacyModal';
import { mustBeChecked } from '@magento/venia-ui/lib/util/formValidators';
import Label from './label';
import { bool, number, object, string } from 'prop-types';
import FormError from '@magento/venia-ui/lib/components/FormError';
import SuccessMessage from '../SuccessMessage';

const CheckboxWithModal = props => {
  const { item, index, isAccountSettingsPage, location } = props;
  const {
    isShowModal,
    handleSubmit,
    modalContent,
    getModalData,
    handleCancel,
    handleClick,
    errors,
    formSubmitMessage,
    label
  } = useCheckboxWithModal({
    checkboxName: 'amgdpr_agreement[' + item.consent_code + ']',
    consentCode: item.consent_code,
    consentFrom: location,
    hasAgreement: item.has_agreement,
    checkboxLabel: item.consent_text
  });

  const getLabel = checkbox => {
    const id = checkbox.consent_id;

    return <Label message={label} id={id} handleClick={getModalData} />;
  };

  const additionalProps =
    item.is_required && !isAccountSettingsPage
      ? { validate: mustBeChecked }
      : {};

  return (
    <Fragment key={index}>
      <Checkbox
        onClick={handleClick}
        id={location + item.consent_id}
        field={`amgdpr_agreement[${item.consent_code}]`}
        label={getLabel(item)}
        {...additionalProps}
      />
      <FormError errors={errors} scrollOnError={false} />
      {location !== 'registration' && (
        <SuccessMessage message={formSubmitMessage} />
      )}
      <Suspense fallbac45k={null}>
        <PrivacyModal
          isOpen={isShowModal}
          onCancel={handleCancel}
          onSubmit={handleSubmit}
          content={modalContent}
          location={location}
        />
      </Suspense>
    </Fragment>
  );
};

CheckboxWithModal.propTypes = {
  item: object,
  index: number,
  isAccountSettingsPage: bool,
  location: string
};

export default CheckboxWithModal;
