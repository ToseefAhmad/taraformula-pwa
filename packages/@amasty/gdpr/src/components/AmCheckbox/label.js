import React, { Fragment } from 'react';
import { FormattedMessage } from 'react-intl';
import LinkButton from '@magento/venia-ui/lib/components/LinkButton';
import { func, number, object } from 'prop-types';

const Label = props => {
  const { message, handleClick, id } = props;

  return (
    <Fragment>
      {message.prevText && (
        <FormattedMessage
          id={`privacy.prevMessage${id}`}
          defaultMessage={message.prevText}
        />
      )}
      {message.text && (
        <LinkButton type="button" onClick={handleClick.bind(this, id)}>
          <FormattedMessage
            id={`privacy.link${id}`}
            defaultMessage={message.text}
          />
        </LinkButton>
      )}
      {message.afterText && (
        <FormattedMessage
          id={`privacy.afterMessage${id}`}
          defaultMessage={message.afterText}
        />
      )}
    </Fragment>
  );
};

Label.propTypes = {
  message: object,
  handleClick: func,
  id: number
};

export default Label;
