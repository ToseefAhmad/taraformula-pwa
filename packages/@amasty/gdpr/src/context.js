import React, { createContext, useContext } from 'react';
import { useAmCheckbox } from './talons/useAmCheckbox';

const amGdprContext = createContext();
const { Provider } = amGdprContext;

const AmGdprContextProvider = props => {
  const { children } = props;

  const contextValue = useAmCheckbox();
  const { error } = contextValue;

  if (error) {
    if (process.env.NODE_ENV !== 'production') {
      console.error(error);
    }

    console.log('Check if Amasty modules has been installed!');
  }

  return <Provider value={contextValue}>{children}</Provider>;
};

export default AmGdprContextProvider;

export const useAmGdprContext = () => useContext(amGdprContext);
