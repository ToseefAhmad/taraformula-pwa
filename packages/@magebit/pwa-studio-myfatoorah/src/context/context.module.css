.spacing {
    composes: mt-4 from global;
}
