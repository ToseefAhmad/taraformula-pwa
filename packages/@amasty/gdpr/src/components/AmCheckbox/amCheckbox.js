import React, { useMemo, useRef } from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import defaultClasses from './amCheckbox.css';
import { useAmGdprContext } from '../../context';
import CheckboxWithModal from './checkboxWithModal';
import { bool, string } from 'prop-types';
import { getVisibleConsents } from '../../../utils/index';

const AmCheckbox = props => {
  const classes = mergeClasses(defaultClasses, props.classes);
  const { isAccountSettingsPage, location } = props;
  const { settings } = useAmGdprContext();
  const consents = useRef(settings);

  const propsArrayFiltered = useMemo(
    () => getVisibleConsents(consents.current, isAccountSettingsPage, location),
    [consents, isAccountSettingsPage, location]
  );

  const checkboxes = propsArrayFiltered.map((checkbox, index) => {
    return (
      <CheckboxWithModal
        key={index}
        item={checkbox}
        index={index}
        isAccountSettingsPage={isAccountSettingsPage}
        location={location}
      />
    );
  });

  if (!checkboxes.length) {
    return null;
  }

  return (
    <div
      className={[
        classes.checkbox,
        location === 'checkout' ? classes.checkout : ''
      ].join(' ')}
    >
      {checkboxes}
    </div>
  );
};

AmCheckbox.propTypes = {
  isAccountSettingsPage: bool,
  location: string
};

export default AmCheckbox;
