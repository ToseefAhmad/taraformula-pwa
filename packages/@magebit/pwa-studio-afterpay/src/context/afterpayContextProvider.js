import { bool, func, instanceOf, node, object } from 'prop-types';
import React, { createContext, useContext } from 'react';

import { useAfterpayCheckout } from '../talons/useAfterpayCheckout';

import { useCartContext } from '@magento/peregrine/lib/context/cart';
import CheckoutError from '@magento/peregrine/lib/talons/CheckoutPage/CheckoutError';

const AfterpayContext = createContext({});
const AfterpayContextProvider = ({
    children,
    handlePlaceOrder,
    checkoutError,
    setCheckoutStep,
    resetReviewOrderButtonClicked,
    isGuestCheckout
}) => {
    const [{ cartId }] = useCartContext();
    const { createAfterpayCheckout, afterpayCheckout } = useAfterpayCheckout({ cartId });

    return (
        <AfterpayContext.Provider
            value={{
                handlePlaceOrder,
                checkoutError,
                setCheckoutStep,
                resetReviewOrderButtonClicked,
                isGuestCheckout,
                createAfterpayCheckout,
                afterpayCheckout
            }}
        >
            {children}
        </AfterpayContext.Provider>
    );
};

AfterpayContextProvider.propTypes = {
    children: node,
    handlePlaceOrder: func,
    checkoutError: instanceOf(CheckoutError),
    setCheckoutStep: func,
    resetReviewOrderButtonClicked: func,
    isGuestCheckout: bool,
    createAfterpayCheckout: func,
    afterpayCheckout: object
};

export default AfterpayContextProvider;
export const useAfterpayContext = () => useContext(AfterpayContext);
