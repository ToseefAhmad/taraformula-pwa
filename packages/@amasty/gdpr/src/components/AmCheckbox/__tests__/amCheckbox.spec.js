import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import AmCheckbox from '../amCheckbox';
import { useAmGdprContext } from '../../../context';

jest.mock('../checkboxWithModal', () => 'CheckboxWithModal');
jest.mock('../../../context', () => {
  const useAmGdprContext = jest.fn(() => {});

  return { useAmGdprContext };
});

test('renders no checkboxes', () => {
  useAmGdprContext.mockReturnValue({
    settings: []
  });

  const tree = createTestInstance(<AmCheckbox isAccountSettingsPage={true} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('renders default amCheckbox', () => {
  useAmGdprContext.mockReturnValue({
    settings: [
      {
        hide_the_consent_after_user_left_the_consent: true
      }
    ]
  });

  const tree = createTestInstance(<AmCheckbox isAccountSettingsPage={true} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
