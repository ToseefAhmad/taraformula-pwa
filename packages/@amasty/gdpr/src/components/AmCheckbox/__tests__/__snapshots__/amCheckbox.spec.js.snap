// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default amCheckbox 1`] = `
<div
  className=" "
>
  <CheckboxWithModal
    index={0}
    isAccountSettingsPage={true}
    item={
      Object {
        "hide_the_consent_after_user_left_the_consent": true,
      }
    }
  />
</div>
`;

exports[`renders no checkboxes 1`] = `null`;
